### DISCUSSION TITLE
Settings if you don't want calulation or regulation by VTherm

### DISCUSSION THREAD

**Discussion by cpktmpkt** [jmcollin78/versatile_thermostat#561] [Q&A] [answered]
I have installed Vtherm for tesing proposes but I'm not sure with the settings.

What I want is the following:
 - using the features presets, presence detection and window open detection (via sensor not calculated by temperature drop)
 - not need/want any regulation or calculation by VTherm because the TRV is connected to an external temperature sensor via zigbee2mqtt with direct binding (both are Aqara devices)

My settings:
 - using `thermostat_over_climate`
 - setting up Aqara external temperature sensor in VTherm, setting
 - setting up the Aqara TRV as underlying wit self-regulation "No auto-regulation"
  

My questions:
 - what are the recommended  settings to avoid regulation by VTherm (if possible)
 - what about regulation threshold and regulation minimum period in underlying?
 - what about cycle duration in mail menu?
 - what does the "use internal temperature of the underlining" option mean?  
 - TPI options take place by using `thermostat_over_climate`?

I'm sure it will help others too.... 

Best regards

cpktmpkt

**Comment by maia** [comment]
In Configuration > Underlyings you'll find the options for self-regulation, you can set it to "none" there. 

Regarding the direct binding, can you explain how it works? I have Aqara temperature sensors and am considering replacing my Eve Thermo (Matter) with z2m devices, probably Sonoff TRVZB (the Aqara have the display on the side not on the top, making them unusable for me). For now I've only used zigbee direct binding for lights, but I wonder how I could control the target temperature with it? Thanks!

**Comment by cpktmpkt** [comment]
Unfortunately it’s not possible to du die binding directly. You need an extension for zigbee2mqtt for that. I followed the instructions here: [link](https://github.com/Koenkk/zigbee2mqtt/discussions/14291)

the advantage of this is you do not need Home Assistant for this so it’s more robust at least if you use an external solution for z2m and not die HA plugin.

> In Configuration > Underlyings you'll find the options for self-regulation, you can set it to "none" there.

that’s all. So all the other settings for regulation do not take place than?

**Reply by maia** [reply]
> Unfortunately it’s not possible to du die binding directly. You need an extension for zigbee2mqtt for that. I followed the instructions here: [link](https://github.com/Koenkk/zigbee2mqtt/discussions/14291)

Thanks, I assume you are using this script/plugin? https://github.com/Koenkk/zigbee2mqtt/discussions/14291#discussioncomment-8254680 . I not yet familiar with plugins/extensions/whateverthenomenclatureis in z2m.

> the advantage of this is you do not need Home Assistant for this so it’s more robust at least if you use an external solution for z2m and not die HA plugin.

True, it's not as robust as zigbee direct bindings though, which I e.g. use for my Hue lights and switches, which even work when the HA server is down. From what I understand it's the zigbee server that sends a payload to the TRV when the external sensor updates, which requires that server to be running – and if it's not, I'm unsure what the TRV will do then. :)
 
> that’s all. So all the other settings for regulation do not take place than?

I think that "regulation threshold" and "regulation minimum period" will be ignored, probably also "cycle duration". This integration has become so complex over the last 18 months that I'm unsure about all the things cycle duration does. 

As for "use internal temperature of the underlining", I can't remember what it does, hopefully someone else can. I wish there way a way to grey out or hide certain parts of the UI when something is disabled.

Attention regarding "Last seen room temperature sensor entity id. Should be datetime sensor", I don't think it can be disabled and it's not yet explained properly in the UI. Basically when the room temperature sensor does not update its temperature for > x minutes, the TRV will go into frost mode and your room temperature will drop to 12 °C. This is an issue for sensors that do not update when the temperature does not change. Somewhere else in the UI there is an option to increase the timeout to a longer time span than the default 60 minutes, I set it to 6 hours and that solved my issue of occasional ice-cold rooms. :)

**Comment by jmcollin78** [comment, marked as answer]
Hello @cpktmpkt 

Here my answers.

> My questions:
> 
> * what are the recommended  settings to avoid regulation by VTherm (if possible)

if you set None in the self-regulation configuration field, you will have no self-regulation from VTherm.

![Capture d’écran 2024-10-16 à 08 03 45](https://github.com/user-attachments/assets/33b2b9ab-09ef-4c78-bda1-0b655666c317)

> * what about regulation threshold and regulation minimum period in underlying?

There are used to limit the number of change done in the TRV and avoid drawning the battery. Set it to 0 and 0 if you want to send all command. This could be pertinent if you select no regulation.

> * what about cycle duration in mail menu?

I'm not sure it is useful if you don't use self-regulation. You can keep it to 5 min.

> * what does the "use internal temperature of the underlining" option mean?

This is useful to add a compensation in the target temperature send. The compensation is the difference between the internal temp of the device and the real external temperature. Don't check it in your case.

> * TPI options take place by using `thermostat_over_climate`?

You should not see this menu in `over_climate`.

**Reply by maia** [reply]
> > * what does the "use internal temperature of the underlining" option mean?
> 
> This is useful to add a compensation in the target temperature send. The compensation is the difference between the internal temp of the device and the real external temperature. Don't check it in your case.

I believe this is a language barrier issue, as even though I've been using vtherm for 18 months and have spent lots of time with it (also here on github), even with your explanation I don't understand what this setting does. Do you happen to have a french text that explains what that checkbox does, that I could run through ChatGPT and let it translate it? Maybe I can come up with a better wording in the UI that explains what that checkbox does. Thanks!

**Reply by jmcollin78** [reply]
Yes English is not my best skill. I agress.

Sure, look in the French README.
```
Compensation de la température interne
Quelque fois, il arrive que le thermomètre interne du sous-jacent (TRV, climatisation, ...) soit tellement faux que l' auto-régulation ne suffise pas à réguler. Cela arrive lorsque le thermomètre interne est trop près de la source de chaleur. La température interne monte alors beaucoup plus vite que la température de la pièce, ce qui génère des défauts dans la régulation. Exemple :

la température de la pièce est 18°, la consigne est à 20°,
la température interne de l'équipement est de 22°,
si VTherm envoie 21° comme consigne (= 20° + 1° d'auto-regulation), alors l'équipement ne chauffera pas car sa température interne (22°) est au-dessus de la consigne (21°)
Pour palier à ça, une nouvelle option facultative a été ajoutée en version 5.4 : Utilisation de la température interne

Lorsqu'elle est activée, cette fonction ajoutera l'écart entre la température interne et la température de la pièce à la consigne pour forcer le chauffage. Dans l'exemple ci-dessus, l'écart est de +4° (22° - 18°), donc VTherm enverra 25° (21°+4°) à l'équipement le forçant ainsi à chauffer.

Cet écart est calculé pour chaque sous-jacent car chacun à sa propre température interne. Pensez à un VTherm qui serait relié à 3 TRV chacun avec sa température interne par exemple.

On obtient alors une auto-régulation bien plus efficace qui évite l'eccueil des gros écarts de température interne défaillante.
```

It is translated in the README (EN) by Google translate.

**Reply by maia** [reply]
Thanks. I suggest to call this feature "Adjust Setpoint for Room vs. TRV Temperature". Here's the text I would use:

––––––
Sometimes, a device’s internal temperature sensor (like in a TRV or AC) can give inaccurate readings, especially if it’s too close to a heat source. This can cause the device to stop heating too soon. For example:

1. target temperature: 20 °C, room temperature: 18 °C, device’s internal sensor: 22 °C
2. If the target temperature is increased to 21 °C, the device won’t heat because it thinks it’s already warm (internal temperature is 22°C).

The **Adjust Setpoint for Room vs. TRV Temperature** feature fixes this by adding the temperature difference between the room and the device’s internal reading to the target. In this case, VTherm would adjust the target to 25°C (21°C + 4°C difference), forcing the device to continue heating.

This adjustment is specific to each device, making the heating system more accurate and avoiding issues from faulty sensor readings.
––––––

PS: I strongly suggest to use ChatGPT 4o (or newer) for translations, as you can give it more context, which will help with the translation. Google Translate will only translate sentence for sentence, not understanding the context at all.

**Reply by maia** [reply]
@jmcollin78 ping, in case you missed this suggestion. in case you like it, please convert it to an issue, it gets lost here in the discussions.

**Reply by jmcollin78** [reply]
Hello @maia 

I update the README. Thank you for your suggestion.

**Reply by maia** [reply]
great, thanks for letting me know! glad I could help.
